tests/rptest/infinite_retention/infinite_retention_test.py

```diff
@@ -0,0 +1,794 @@
+            self.logger.warn(
+                "No uploaded segments found! Check test parameters!")
+        else:
+            self.logger.info(f"Max uploaded ts = {stats['uploaded_ts']}")
+
+            lag_seconds = (stats['local_ts'] - stats['uploaded_ts']) / 1000.0
+            config_interval = (self.params.manifest_upload_interval +
+                               (self.params.segment_size / actual_byte_rate))
+            self.logger.info(f"Upload lag: {lag_seconds}s, "
+                             f"Upload interval: {config_interval}")
+            chk.add_arg("lag_seconds", lag_seconds)
+            chk.add_arg("config_interval", config_interval)
+            chk.add_check(
+                "Measure how far behind the tiered storage uploads are: "
+                "success condition should be that they are within some "
+                "time range of the most recently produced data",
+                "lag_seconds < config_interval",
+            )
+
+        # Wait for all uploads to complete: this should take roughly
+        # segment_max_upload_interval_sec plus manifest_max_upload_interval_sec
+        quiesce_uploads(self.redpanda, [self.topic],
+                        timeout_sec=self.params.manifest_upload_interval +
+                        self.params.segment_upload_interval)
+
+        # Check manifest upload metrics:
+        #  - we should not have uploaded the manifest more times
+        #    then there were manifest upload intervals in the runtime.
+        self.logger.info(
+            f"Upload counts: {stats['manifest_uploads']} manifests, "
+            f"{stats['segment_uploads']} segments")
+        chk.add_arg("manifest_uploads", stats['manifest_uploads'])
+        chk.add_arg("segment_uploads", stats['segment_uploads'])
+        chk.add_check(
+            "Non-zero manifest uploads",
+            "manifest_uploads > 0",
+        )
+        chk.add_check(
+            "Non-zero segment uploads",
+            "segment_uploads > 0",
+        )
+
+        # +3 because:
+        # - 1 empty upload at start
+        # - 1 extra upload from runtime % upload interval
+        # - 1 extra upload after the final interval_sec driven uploads
+        expect_manifest_uploads = (
+            (int(produce_duration) // self.params.manifest_upload_interval) +
+            3)
+        self.logger.info(f"Manifests uploads: {stats['manifest_uploads']}, "
+                         f"Expected not less than: {expect_manifest_uploads}")
+        chk.add_arg("produce_duration", int(produce_duration))
+        chk.add_arg("manifest_upload_interval",
+                    self.params.manifest_upload_interval)
+        chk.add_check(
+            "For spillover active tests manifest uploads should be "
+            "significant",
+            "manifest_uploads > produce_duration // manifest_upload_interval + 3",
+        )
+        # Do all required checks with Redpanda active
+        chk.conduct_checks()
```

Comment:
Even if this test is primarily about producers, it should run a consumer at the end to read the data back and ensure  it is all accessible and that the same number of messages are  consumed as produced.

Reply:
This is tricky here. Actual SeqConsumer that was used in the latest run going loops in message cosumption after fixes and we can't get actually consumed message counts that easily. On the other hand, the goal here is to consume messages the fastest and bandwidth wise as full as possible to introduce crowded service use.